schemas/database/001_initial.sql

```diff
@@ -8,3 +8,23 @@ CREATE TABLE transactions (
     raw         BYTEA                    NOT NULL,
     hash        BYTEA                        NULL
 );
+
+CREATE TABLE log_fetches
+(
+    id          SERIAL PRIMARY KEY NOT NULL,
+    timestamp   TIMESTAMP          NOT NULL,
+    first_block BIGINT             NOT NULL,
+    last_block  BIGINT             NOT NULL,
+);
+
+CREATE TABLE logs
+(
+    block_index         BIGINT         NOT NULL,
+    transaction_index   BIGINT         NOT NULL,
+    log_index           BIGINT         NOT NULL,
+    fetch_id            INT            NOT NULL REFERENCES log_fetches (id),
+    group_id            NUMERIC(78, 0) NOT NULL,
+    identity_commitment NUMERIC(78, 0) NOT NULL,
+    root                NUMERIC(78, 0) NOT NULL,
+    UNIQUE (block_index, transaction_index, log_index)
```

Comment:
This is good. Just want to note that using block-number alone is not necessarily sufficient. If a 'reorg' happens the last N blocks get overwritten by a different set of blocks with different transactions/log events. This used to be a lot more common, but with the PoS merge it's about once a week:

https://etherscan.io/blocks_forked

One simple solution that works here is to never sync to the very latest block, but always stay N blocks behind the latest. Where N is the maximum "reorg-depth" that you want to support. Usually `10` is chosen as a conservative maximum.